docs/content/configuration/index.md

```diff
@@ -1251,7 +1251,11 @@ These Historical configurations can be defined in the `historical/runtime.proper
 |`druid.segmentCache.infoDir`|Historical nodes keep track of the segments they are serving so that when the process is restarted they can reload the same segments without waiting for the Coordinator to reassign. This path defines where this metadata is kept. Directory will be created if needed.|${first_location}/info_dir|
 |`druid.segmentCache.announceIntervalMillis`|How frequently to announce segments while segments are loading from cache. Set this value to zero to wait for all segments to be loaded before announcing.|5000 (5 seconds)|
 |`druid.segmentCache.numLoadingThreads`|How many segments to drop or load concurrently from from deep storage.|10|
-|`druid.segmentCache.numBootstrapThreads`|How many segments to load concurrently from local storage at startup.|Same as numLoadingThreads|
+|`druid.coordinator.loadqueuepeon.curator.numCreateThreads`|Number of threads creating zk nodes corresponding to segments that need to be loaded or dropped.|10|
+|`druid.coordinator.loadqueuepeon.curator.numCallbackThreads`|Number of threads for executing callback actions associated with loading or dropping of segments.|2|
+|`druid.coordinator.loadqueuepeon.curator.numMonitorThreads`|Number of threads to use for monitoring deletion of zk nodes|1|
+|`druid.coordinator.curator.create.zknode.batchSize`|Number of zk nodes to create in one iteration.|5000|
+|`druid.coordinator.curator.create.zknode.repeatDelay`|Delay before creating next batch of zk nodes|PT1M|
```

Comment:
I wonder how should the operator calculate the delay and batch size? I think it's complex right now.
For example I understand that
`delay*batch_size*avg_segment_size<=node_network_throughput`
But I can't figure out what number to select here, as they depend on each other.  Probably there should be one parameter like `max_node_network_throughput` and the Peon keeps sending load request until the limit is hit, then the Peon delays requests until the quota is replenished.

Reply:
Good question, @egor-ryashin . I ended up following the pattern we have for http based segment loading. The general  idea was to allow historicals long enough to load the segment before coordinator tries to recreate the zookeeper node for it. Keep in mind that segment loading is not purely I/O. It also involves decompressing the segment files and mapping them to off-heap memory. But yes, the general formula to come up with the numbers is what you suggested:

```(batch_size * average segment_size)/delay <= factor * node_network_throughput```

IMHO, having separate parameters for batch size and delay makes more sense since it would force users to think more about how values to set instead of just setting a number like node_network_throughput. I can also update the documentation with the above formula to aid users. Of course, most of the users won't be configuring this. So having sane default values is important.

Reply:
The problem with the formula is that `avg_segment_size` can vary over time, I used the formula merely to find out if we understand each other well. 
> having separate parameters for batch size and delay makes more sense since it would force users to think more about how values to set instead of just setting a number like node_network_throughput.

The thing is, `batch size` and `delay` indirectly depend on the data being processed, if segments become larger the Druid user should somehow calculate the average segment size then reconfigure the Druid cluster and hope that the average is not skewed in some way, while having a `max_network_throughput` parameter the user can configure once and forget.

Should I say that the Druid user has to know `max_network_throughput` using both aforementioned implementations, but with the second imlementation the user has to do much less maintenance work.

Reply:
If we adopt max_network_throughput, then we would need to measure the rate at which we are adding segments to the processing queue. It is doable and there are rate limiter implementations out there including in Guava. Having said that, not all of the max_network_throughput configured by user can be directly used for rate limiting purposes. Time is spent not only downloading segments, but also in decompressing and memory mapping. So the real rate to limit would be a factor of the throughput. Would that mean we would need to make the factor configurable (which isn't too bad really, just wanted to call it out though). 

Any other feedback you have @egor-ryashin? . I would like to get this pull request in before it goes stale beyond recovery :).

Reply:
Looks like there already is a config ```maxSegmentsInNodeLoadingQueue``` which controls how many segments queued up for a historical node to load. 
http://druid.io/docs/latest/configuration/index.html#dynamic-configuration

The default value of this dynamic config is 0 meaning there is no queue and the coordinator is going to assign segments for historicals. In practice though, I have found that this default value ends up causing clusters to be a bit unbalanced. In our production clusters, where we have this pull request, we have set this value to 1000 (which is what the config documentation also recommends).

What this means is that, we don't need to worry about batching or control the rate at which historicals are going to download segments. The above config is already providing that behavior. I am going to get rid off them.